### PR TITLE
JBPM-5336 - Support for rolling updates in kie server

### PR DESCRIPTION
**Pull request by mswiderski**

### CODE REVIEW

**Comment by mswiderski** [comment]
@krisv @etirelli @sutaakar whenever you find some time could you please have a look?

some details can be found in [this article](http://mswiderski.blogspot.com/2016/09/improved-container-handling-and-updates.html)

UI changes to support alias for container is https://github.com/droolsjbpm/kie-wb-common/pull/491

**Comment by etirelli** [comment]
Great job @mswiderski ! I will take a look at the code changes, but this is a very useful feature.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            assertEquals(CONTAINER_ID_101, processInstance.getContainerId());
+
+            tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertEquals(1, tasks.size());
+            task = tasks.get(0);
+            assertEquals(CONTAINER_ID_101, task.getContainerId());
+
+            taskClient.completeAutoProgress(CONTAINER_ALIAS, task.getId(), USER_YODA, null);
+        } finally {
+            if (processInstanceIdV1 != null) {
```
Comment (rsynek) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java]: I think aborting processes could be done in a separate method with variable number of parameters, or even better - moved to a `@Before` method to decouple test scenario from clean up code and avoid repetition.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+            assertEquals(CONTAINER_ID, processInstance.getContainerId());
+
+            tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertEquals(1, tasks.size());
+            task = tasks.get(0);
+            assertEquals(CONTAINER_ID, task.getContainerId());
+
+            taskClient.completeAutoProgress(CONTAINER_ALIAS, task.getId(), USER_YODA, null);
+        } finally {
+            if (processInstanceIdV1 != null) {
```
Comment (rsynek) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java]: The same as for `testStartProcessInDifferentDeploymentWithAlias()`

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java (hunk 1 above)
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java]: It is actually done, see JbpmKieServerBaseIntegrationTest.

**Comment by mswiderski** [comment]
thanks @rsynek @sutaakar for comments, I made the adjustments

**Pull request merged**